modules/exploits/multi/http/joomla_http_header_rce.rb

```diff
@@ -0,0 +1,183 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Joomla HTTP Header Unauthenticated Remote Code Execution',
+      'Description'    => %q{
+          Joomla suffers from an unauthenticated remote code execution that affects all versions from 1.5.0 to 3.4.5.
+          By storing user supplied headers in the databases session table it's possible to truncate the input
+          by sending an UTF-8 character. The custom created payload is then executed once the session is read
+          from the databse. You also need to have a PHP version before 5.4.45 (including 5.3.x), 5.5.29 or 5.6.13.
+          In later versions the deserialisation of invalid session data stops on the first error and the
+          exploit will not work. The PHP Patch was included in Ubuntu versions 5.5.9+dfsg-1ubuntu4.13 and
+          5.3.10-1ubuntu3.20 and in Debian in version 5.4.45-0+deb7u1.
+      },
+      'Author'	=>
+        [
+          'Marc-Alexandre Montpas', # discovery
+          'Christian Mehlmauer' # metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['CVE', '2015-8562'],
+          ['URL', 'https://blog.sucuri.net/2015/12/joomla-remote-code-execution-the-details.html'],
+          ['URL', 'https://blog.sucuri.net/2015/12/remote-command-execution-vulnerability-in-joomla.html'],
+          ['URL', 'https://developer.joomla.org/security-centre/630-20151214-core-remote-code-execution-vulnerability.html'],
+          ['URL', 'https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=http%3A%2F%2Fdrops.wooyun.org%2Fpapers%2F11330'],
+          ['URL', 'https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=http%3A%2F%2Fwww.freebuf.com%2Fvuls%2F89754.html'],
+          ['URL', 'https://bugs.php.net/bug.php?id=70219']
+        ],
+      'Privileged'     => false,
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['Joomla', {}]],
```

Comment:
Just tried with PHP versions but this is getting too big:
```
Exploit target:

   Id  Name
   --  ----
   0   Joomla < 3.4.6, PHP < 5.4.45 (including 5.3.x), < 5.5.29 and < 5.6.13, on Ubuntu < 5.5.9+dfsg-1ubuntu4.13 and < 5.3.10-1ubuntu3.20, on Debian < 5.4.45-0+deb7u1
```

I think the Joomla Version is enough?